### PR TITLE
Break-out perspective for better IE11 support.

### PR DESCRIPTION
**Pull request by jdalton**
This PR is an attempt to resolve the IE perspective issue described in #476.

### CODE REVIEW

**Comment by FagnerMartinsBrack** [comment]
If this fixes the problem, is there any way to build a failing test for IE to prevent regression?

**Comment by jdalton** [comment]
<table>
<tr>
<th>before</th><th>after</th>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
</table>

**Comment by jdalton** [comment]
Based on the before and after it looks like it fixes the issue without regressing others (I checked Chrome at least) and without relying the `ms-` prefix.

I'll look into adding a test.

**Comment by FagnerMartinsBrack** [comment]
Eventually, we need to get visual testing, but while we're not there if we can fail the test for IE by trying to retrieve the perspective data and then fix with this change then it would be a considerably decent regression test given the circumstances...

**Comment by jdalton** [comment]
Yep, no worries. I'll cook up a test and a code comment prolly won't hurt either.

**Comment by jdalton** [comment]
Tests added.

File: js/impress.js [hunk 1]
```diff
@@ -133,11 +133,6 @@
         return " scale(" + s + ") ";
     };
 
-    // `perspective` builds a perspective transform string for given data.
-    var perspective = function( p ) {
-        return " perspective(" + p + "px) ";
-    };
-
```
Comment (FagnerMartinsBrack) [inline js/impress.js]: I agree it's a good thing remove the function here since it doesn't have good cohesion (`perspective()` says nothing really). Sometimes no abstraction is better than the wrong abstraction 👍

File: test/core_tests.js [hunk 2]
```diff
@@ -37,9 +37,10 @@
     });
 
     QUnit.test( "Attributes", function( assert ) {
-      assert.expect( 9 );
+      assert.expect( 10 );
 
       var actual, expected;
+      var root = document.querySelector( "div#impress" );
```
Comment (FagnerMartinsBrack) [inline test/core_tests.js]: Is there a reason for using `div#impress` instead of just `impress` in the selector? I suspect this tried to keep consistency but the line below seems to have selected the `html` by mistake, is that right?

File: test/core_tests.js [hunk 3]
```diff
@@ -78,6 +79,11 @@
       expected = "ease-in-out";
       assert.strictEqual( actual, expected, "canvas.style.transitionTimingFunction initialized correctly" );
 
+      // Ensure we specify perspective independent of transform for IE 11
```
Comment (FagnerMartinsBrack) [inline test/core_tests.js]: Maybe if we change the message for something like `"root.style.perspective should be set explicitly for IE 11"` then we wouldn't need the comment and it would show the "why" when the test fails in the console.

The other tests are like "`n` initialized correctly" because we don't really had a reason for testing that attribute, it was just to have some blind coverage somewhere (which actually worked since when we added SauceLabs the tests failed in Safari xD).

File: test/core_tests.js [hunk 4]
```diff
@@ -78,6 +79,11 @@
...
+      actual = root.style.perspective;
+      expected = "";
+      assert.notStrictEqual( actual, expected, "root.style.perspective initialized correctly" );
```
Comment (FagnerMartinsBrack) [inline test/core_tests.js]: It looks like this test wouldn't fail only in IE but just serve as a verification that if we remove the `perspective` property it will fail in any browser, is that right?

Comment (jdalton) [inline test/core_tests.js]: Correct.

File: js/impress.js [hunk 5]
```diff
@@ -205,9 +200,7 @@
         height: 768,
         maxScale: 1,
         minScale: 0,
-
```
Comment (jdalton) [inline js/impress.js]: Noticed a format inconsistency.

File: test/core_tests.js (hunk 2 above)
Comment (jdalton) [inline test/core_tests.js]: Ya I did it for consistency with other references.

File: js/impress.js (hunk 5 above)
Comment (FagnerMartinsBrack) [inline js/impress.js]: Can JSCS or JSHINT handle this? It might be more useful if we could automate this kind of stuff to reduce the chances it's spotted in unrelated PRs.

File: test/core_tests.js [hunk 6]
```diff
@@ -49,7 +50,7 @@
       expected = "rotateZ(0deg) rotateY(0deg) rotateX(0deg) translate3d(1000px, 0px, 0px)";
       assert.strictEqual( actual, expected, "canvas.style.transform initialized correctly" );
 
-      // Normalize result for IE 11 and Safari.
+      // Normalize result for IE 11 and Safari
```
Comment (FagnerMartinsBrack) [inline test/core_tests.js]: Can JSCS or JSHINT handle this? It might be more useful if we could automate this kind of stuff to reduce the chances it's spotted in unrelated PRs.

Comment (FagnerMartinsBrack) [inline test/core_tests.js]: (I wonder if it's not more useful to do this kind of change in a separate PR, though?)

**Comment by jdalton** [comment]
Updated.
